drivers/vulkan/vulkan_context.cpp

```diff
@@ -2767,6 +2783,15 @@ void VulkanContext::set_vsync_mode(DisplayServer::WindowID p_window, DisplayServ
 }
 
 VulkanContext::VulkanContext() {
+	frame_count = uint8_t(GLOBAL_DEF_RST("display/window/vsync/buffer_count", 2u));
+	// TODO: In theory it should be possible to have swapchain_desired_count per window.
+	// But it may complicate their management.
+	swapchain_desired_count = uint8_t(GLOBAL_DEF_RST("display/window/vsync/swapchain_count", 3u));
+
+	CRASH_COND_MSG(frame_count > MAX_FRAME_LAG, "Frame count out of bounds.");
+	CRASH_COND_MSG(frame_count < 1, "Frame count out of bounds.");
+	CRASH_COND_MSG(swapchain_desired_count < 1, "Frame count out of bounds.");
```

Comment:
You probably want to make the error message here match the condition to describe that swap chain desired count is the problem.

Reply:
Reworded as suggested. I agree with all of it. It's much more clear to the user.

The old wording was from the initial version of the code which was not taking the input directly from `GLOBAL_DEF_RST`.

Reply:
Reworded as suggested. I agree with all of it. It's much more clear to the user.

The old wording was from the initial version of the code which was not taking the input directly from `GLOBAL_DEF_RST`.